app/src/main/java/com/example/calculator/base/MainActivity.kt

```diff
@@ -0,0 +1,148 @@
+package com.example.calculator.base
+
+import android.content.Context
+import android.content.Intent
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import com.example.calculator.R
+import com.example.calculator.view.calculator.CalculatorFragment
+
+class MainActivity : BaseActivity(R.layout.activity_main) {
+    
+    private var isBack = false
+    
+    private val fragmentManager by lazy {
+        supportFragmentManager
+    }
+    
+    var onBackPressedListener: OnBackPressedListener? = null
+    
+    override fun onCreatedView() {
+        
+        replaceFragment(CalculatorFragment())
+        
+        fragmentManager.addOnBackStackChangedListener {
+            if (isBack) {
+                val backStackEntryCount: Int = fragmentManager.backStackEntryCount
+                if (backStackEntryCount > 0 && backStackEntryCount <= fragmentManager.fragments.size) {
+                    try {
+                        val fragment: Fragment = fragmentManager.fragments[backStackEntryCount - 1]
+                        fragment.onResume()
+                    }
+                    catch (e: Exception) {
+                    }
+                }
+            }
+        }
+    }
+    
+    override fun onBackPressed() {
+        if (onBackPressedListener != null) {
+            onBackPressedListener?.doBack()
+        }
+    }
+    
+    fun addFragment(fragment: Fragment, tag: String? = "", animation: Int? = 1) {
+        when (animation) {
+            0 -> {
```

Comment:
hard code 
Đoạn này a có thể dùng EnumClass 
```suggestion
            AnimationTypes.NON -> {
```